src/ticket-search/ticket-search/Services/RsiServices/RsiTicketsService.cs

```diff
@@ -7,23 +10,27 @@ namespace Gov.TicketSearch.Services.RsiServices
     public class RsiTicketsService : TicketsService
     {
         private readonly IRsiRestApi _rsiApi;
-        public RsiTicketsService(IRsiRestApi rsiApi)
+        public RsiTicketsService(IRsiRestApi rsiApi, ILogger<TicketsService> logger) :base(logger)
         {
-            _rsiApi = rsiApi;
+            _rsiApi = rsiApi ?? throw new ArgumentNullException(nameof(rsiApi));
         }
 
         protected override async Task<RawTicketSearchResponse> GetTicket(string ticketNumber, string time, CancellationToken cancellationToken)
         {
+            _logger.LogDebug("send ticket search request to Rsi.");
             var result = await _rsiApi.GetTicket(
                 new GetTicketParams { TicketNumber = ticketNumber, Prn = "10006", IssuedTime = time.Replace(":", "") },
                 cancellationToken);
+            _logger.LogDebug("get ticket info from Rsi : {result}", JsonSerializer.Serialize(result));
             return result;
 
         }
 
         protected override async Task<Invoice> GetInvoiceAsync(string invoiceNumber, CancellationToken cancellationToken)
         {
-            return await _rsiApi.GetInvoice(invoiceNumber: invoiceNumber, cancellationToken: cancellationToken);
+            var invoice = await _rsiApi.GetInvoice(invoiceNumber: invoiceNumber, cancellationToken: cancellationToken);
+            _logger.LogDebug("get invoice from Rsi : {invoice}", JsonSerializer.Serialize(invoice));
```

Comment:
We do not need to serialize items as json for logging. The structured logging will do that.

Reply:
Thanks. I got improved again.
fixed.